### PR TITLE
Add language.deprecated.symbolLiterals import

### PR DESCRIPTION
**Pull request by odersky**
Allow symbol literals if a language import

    import language.deprecated.symbolLiterals

is given. This is useful for legacy software that uses
symbol literals. A typical case are Spark notebooks, where
column names are expressed with symbol literals.

### CODE REVIEW

**Comment by odersky** [comment]
This PR addresses https://github.com/lampepfl/dotty-feature-requests/issues/182.

**Review by bishabosha — APPROVED** [review]
LGTM

File: compiler/src/dotty/tools/dotc/parsing/Parsers.scala [hunk 1]
```diff
@@ -1193,17 +1193,17 @@ object Parsers {
...
-              in.sourcePos())
-            if migrateTo3 then
-              patch(source, Span(in.offset, in.offset + 1), "Symbol(\"")
-              patch(source, Span(in.charOffset - 1), "\")")
+          else
+            if !symbolLiteralsEnabled(using languageImportContext) then
+              report.errorOrMigrationWarning(
+                em"""symbol literal '${in.name} is no longer supported,
+                    |use a string literal "${in.name}" or an application Symbol("${in.name}") instead,
+                    |or enclose in braces '{${in.name}} if you want a quoted expression.""",
```
Comment (bishabosha) [inline compiler/src/dotty/tools/dotc/parsing/Parsers.scala:1201]: Should this error mention the import?

Comment (odersky) [reply compiler/src/dotty/tools/dotc/parsing/Parsers.scala:1201]: yes, I think that would be a good idea

**Pull request merged**